rococo/data/mongodb.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any, Dict, List, Tuple, Union
+from rococo.data.base import DbAdapter
+from pymongo import MongoClient
+
+
+class MongoDBAdapter(DbAdapter):
+    """MongoDB adapter for interacting with MongoDB."""
+
+    def __init__(self, mongo_uri: str, mongo_database: str):
+        self.client = MongoClient(mongo_uri)
+        self.db_name = mongo_database
+        self.db = self.client.get_database(mongo_database)
+
+    def __enter__(self):
+        """Context manager entry point for preparing DB connection."""
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.db = None
+
+    def move_entity_to_audit_table(self, table, entity_id):
+        """Executes a query to move entity to audit table."""
+        entity = self.get_one(table, "entity_id", {"entity_id": entity_id})
+        self.db[f"{table}_audit"].insert_one(entity)
+        return entity
+
+    def _call_db(self, function_name, *args, **kwargs):
+        """Calls a function specified by function_name argument in MongoDB connection passing forward args and kwargs."""
+        if not self._db:
+            raise Exception("No connection to MongoDB.")
+        return self._event_loop.run_until_complete(getattr(self._db, function_name)(*args, **kwargs))
```

Comment:
Since `_call_db`  it is not used anywhere in the pr, and is not a part of a public interface, let's not add it.